service/api_task.go

```diff
@@ -411,25 +439,40 @@ func assignNextAvailableTask(taskQueue *model.TaskQueue, taskQueueService model.
 				"host":    currentHost.Id,
 				"message": "skipping task because of disabled project",
 			})
-			// Dequeue the task so we don't get it on another iteration of the loop.
-			if err = taskQueue.DequeueTask(nextTask.Id); err != nil {
-				return nil, errors.Wrapf(err,
-					"error pulling task with id %s from queue for distro %s",
-					nextTask.Id, nextTask.DistroId)
-			}
+
+			grip.Warning(message.WrapError(taskQueue.DequeueTask(nextTask.Id), message.Fields{
```

Comment:
It seems wrong not to return in this case or the one below, since if we can't dequeue the task, we'll go through this loop again, potentially entering an infinite loop if we can't ever dequeue the task. It's safer to exit, since this loop is not over tasks in the group, but instead `for taskQueue.Length() != 0`.

Reply:
As discussed: we'll leave this behaviour as is, for now.